Python2 cleanup

sbesson opened the pull request:
Discovered during the review of https://github.com/ome/omero-prometheus-tools/pull/13

Deployments which have been upgraded from `ome.omero_prometheus_exporter 0.2.x` to `ome.omero_prometheus_exporter 0.3.x` still contain the former Python 2 directories alongside the Python 3 virtual environment. This can lead to confusion.

3de5843dba33714e83313e2054f2378dd1c0c992 proposes to address this by cleaning the Python 2 directories as part of the role
aaa4d47966b20dc2b5d129bd82d229c049f599dd also upgrades the version of `omero_prometheus_tools` and updates the README to make it clear that the library version can be updated via role variable

joshmoore reviewed (commented):
Changes all make sense to me. Shall we try it against pub-omero? (Or have you run it somewhere else?)

sbesson commented:
> Shall we try it against pub-omero? 

Just executed the `omero-monitoring-agents.yml` playbook

<details>
 <summary>Playbook execution</summary>

```
PLAY [omero-server] ********************************************************************************************************************************

TASK [Gathering Facts] *****************************************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_jmx : prometheus jmx | create directories] ************************************************************************************
ok: [pub-omero.openmicroscopy.org] => (item=/opt/prometheus/jars)
ok: [pub-omero.openmicroscopy.org] => (item=/etc/prometheus)

TASK [ome.prometheus_jmx : prometheus jmx | download jar] ******************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_jmx : prometheus jmx | symlink jar] *******************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_jmx : prometheus jmx | configuration file] ************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : Ensure postgres users exists] **************************************************************************************
skipping: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | create directories] **************************************************************************
ok: [pub-omero.openmicroscopy.org] => (item=/opt/prometheus/postgres)
ok: [pub-omero.openmicroscopy.org] => (item=/etc/prometheus)

TASK [ome.prometheus_postgres : prometheus postgres | download] ************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | install postgres-exporter] *******************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | symlink postgres-exporter] *******************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | configuration file] **************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | server systemd service] **********************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_postgres : prometheus postgres | enable and start server systemd] *************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_common : omero-common | create parent directory] ***********************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_common : omero-common | check if services installed] *******************************************************************************
ok: [pub-omero.openmicroscopy.org] => (item=/etc/systemd/system/omero-server.service)
ok: [pub-omero.openmicroscopy.org] => (item=/etc/systemd/system/omero-web.service)
ok: [pub-omero.openmicroscopy.org] => (item=/usr/lib/systemd/system/nginx.service)

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | install python3] *****************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero web | install python3] *********************************************************************************
skipping: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | create system user] **************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | setup virtualenv3] ***************************************************************
changed: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | install omero-prometheus-tools] **************************************************
changed: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | delete Python 2 installation directories] ****************************************
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/prometheus-omero-tools/bin",
-    "state": "directory"
+    "state": "absent"
 }

changed: [pub-omero.openmicroscopy.org] => (item=bin)
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/prometheus-omero-tools/etc",
-    "state": "directory"
+    "state": "absent"
 }

changed: [pub-omero.openmicroscopy.org] => (item=etc)
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/prometheus-omero-tools/include",
-    "state": "directory"
+    "state": "absent"
 }

changed: [pub-omero.openmicroscopy.org] => (item=include)
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/prometheus-omero-tools/lib",
-    "state": "directory"
+    "state": "absent"
 }

changed: [pub-omero.openmicroscopy.org] => (item=lib)
--- before
+++ after
@@ -1,4 +1,4 @@
 {
     "path": "/opt/prometheus-omero-tools/lib64",
-    "state": "link"
+    "state": "absent"
 }

changed: [pub-omero.openmicroscopy.org] => (item=lib64)

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | configure service] ***************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | install service] *****************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_prometheus_exporter : omero prometheus exporter | enable service] ******************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [omero-server prometheus jmx agents] **********************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

RUNNING HANDLER [ome.omero_prometheus_exporter : restart prometheus-omero-exporter] ****************************************************************
changed: [pub-omero.openmicroscopy.org]

PLAY [omero-web] ***********************************************************************************************************************************

TASK [ome.omero_common : omero-common | create parent directory] ***********************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_common : omero-common | check if services installed] *******************************************************************************
ok: [pub-omero.openmicroscopy.org] => (item=/etc/systemd/system/omero-server.service)
ok: [pub-omero.openmicroscopy.org] => (item=/etc/systemd/system/omero-web.service)
ok: [pub-omero.openmicroscopy.org] => (item=/usr/lib/systemd/system/nginx.service)

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | install] *********************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | configure] *******************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | configure gunicorn] **********************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | service.d directory] *********************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | service.d file] **************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.omero_web_django_prometheus : omero-web django prometheus | stats dir] *******************************************************************
ok: [pub-omero.openmicroscopy.org]

PLAY [monitored] ***********************************************************************************************************************************

TASK [ome.prometheus_node : prometheus node | create directories] **********************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_node : prometheus node | download node-exporter] ******************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_node : prometheus node | install node-exporter] *******************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_node : prometheus node | symlink node-exporter] *******************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_node : prometheus node | systemd service] *************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.prometheus_node : prometheus node | enable and start systemd service] ********************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.selinux_utils : selinux | check exists] **************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.selinux_utils : selinux | check enabled] *************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.selinux_utils : system packages | set selinux variable] **********************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [ome.selinux_utils : system packages | install selinux utilities] *****************************************************************************
[DEPRECATION WARNING]: Invoking "yum" only once while using a loop via squash_actions is deprecated. Instead of using a loop to supply multiple 
items and specifying `name: "{{ item }}"`, please use `name: ['libselinux-python', 'libsemanage-python', 'policycoreutils-python']` and remove the 
loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ok: [pub-omero.openmicroscopy.org] => (item=['libselinux-python', 'libsemanage-python', 'policycoreutils-python'])

TASK [nginx selinux allow network connect] *********************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [Create monitoring htpasswd] ******************************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

TASK [Create nginx proxy for prometheus exporters] *************************************************************************************************
ok: [pub-omero.openmicroscopy.org]

PLAY [omero-web] ***********************************************************************************************************************************

TASK [Create nginx proxy for prometheus web exporters] *********************************************************************************************
ok: [pub-omero.openmicroscopy.org]

PLAY RECAP *****************************************************************************************************************************************
pub-omero.openmicroscopy.org : ok=46   changed=4    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0   

(idr-ansible) (base) s
```
</details>

joshmoore commented:
```
[jamoore@pub-omero ~]$ cd /opt/prometheus-omero-tools/
[jamoore@pub-omero prometheus-omero-tools]$ ls -ltra
total 0
drwxr-xr-x. 6 root root  77 Dec 14  2017 ..
drwxr-xr-x. 6 root root 111 Nov 26 13:29 venv3
drwxr-xr-x. 3 root root  19 Nov 26 13:29 .
```

:+1:

(I'll not mention the name of the directory.... :wink:)

sbesson commented:
> (I'll not mention the name of the directory.... 😉)

Heh. Feel free to create an issue but I am not proposing to fix the naming consistency as part of this PR.

Objections to releasing as `0.3.3` ?

joshmoore reviewed (approved):
No objections.
